11_circular_lkoenig/orientation_polar_chart.R

```diff
@@ -0,0 +1,99 @@
+library(tidyverse)
+library(patchwork)
+library(nhdplusTools)
+library(sf)
+library(ggspatial)
+library(showtext)
+
+# Source helper functions
+source("11_circular_lkoenig/src/orientation_helpers.R")
+
+# Set up fonts for plots
+font_add_google('Source Sans Pro', regular.wt = 300, bold.wt = 700)
+showtext_auto()
+
+# Define huc8 sub-basins
+# note: HUC8 subsets from the watershed boundary dataset (WBD) were used for 
+# convenience for this river orientation visualization, to represent a reasonable, 
+# "medium" scale of analysis and so that a user wouldn't need to have the national
+# hydrography dataset (NHDPlusv2) downloaded locally. The selected HUC8 id's 
+# represent 'whole' watersheds/river networks, but not all HUC8 subsets do (i.e.,
+# they may represent just a portion of a watershed). 
+huc8_tbl <- tibble(huc8_id = c("05090202","18040008","02070001","17110014",
+                               "03120002","10030205","17070204","10200103",
+                               "04110002","15050202","08080101","02040101"),
+                   huc8_name = c("Little Miami","Merced","South Branch Potomac",
+                                 "Puyallup","Upper Ochlockonee","Teton",
+                                 "Lower John Day", "Middle Platte-Prairie",
+                                 "Cuyahoga","Upper San Pedro","Atchafalaya",
+                                 "Upper Delaware"))
+
+# Fetch NHDv2 flowlines for each huc8 basin
+flines <- lapply(huc8_tbl$huc8_id, fetch_flowlines)
+
+# Estimate channel orientation (i.e., azimuth) for each huc8 basin
+# note that this step is taking a while to run, ~10 min?
+flines_azimuth <- lapply(flines, function(x){
+  az_df <- x %>%
+    split(., 1:length(.$geometry)) %>%
+    purrr::map_dfr(~mutate(., azimuth = calc_azimuth_circ_mean(.)))
+  return(az_df)
+})
```

Comment:
I am getting an error running this:
```
! Problem with `mutate()` column `azimuth`.
ℹ `azimuth = calc_azimuth_circ_mean(.)`.
x bearing needs at least 2 points

Reply:
Ah, bummer. I haven't been able to reproduce this error on my USGS laptop (windows) or my personal computer (mac). But I suspect there was a glitch when downloading the flowlines and that one of the objects in `flines` is an empty tibble with zero rows. If that's the case, I would expect that re-running the code that defines `flines` might get around the error. But this error points to the need to add code to `fetch_flowlines` to deal with fragility there.